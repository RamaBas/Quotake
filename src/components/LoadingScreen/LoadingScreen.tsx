import React from 'react';
import { Box, Typography } from '@mui/material';
import 'animate.css/animate.min.css';

const LoadingScreen: React.FC = () => {
  return (
    <Box
      data-testid="loading-screen"
      sx={{
        position: 'fixed',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'primary.main',
        zIndex: 9999
      }}
    >
      <Typography
        variant="h1"
        sx={{
          color: 'white',
          fontSize: '10rem',
          fontWeight: 'bold'
        }}
        className="animate__animated animate__flip"
      >
        Q!
      </Typography>
    </Box>
  );
};

export default LoadingScreen;