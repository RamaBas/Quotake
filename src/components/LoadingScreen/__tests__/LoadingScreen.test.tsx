import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import LoadingScreen from '../LoadingScreen';

describe('LoadingScreen Component', () => {
  it('renders correctly with fixed positioning', () => {
    render(<LoadingScreen />);
    
    const loadingBox = screen.getByTestId('loading-screen');
    expect(loadingBox).toBeInTheDocument();
  });
});