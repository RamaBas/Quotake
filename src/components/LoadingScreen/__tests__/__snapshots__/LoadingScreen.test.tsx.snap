// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen Component renders correctly with fixed positioning 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1qm9j7u"
    data-testid="loading-screen"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 animate__animated animate__flip css-8vrb60-MuiTypography-root"
    >
      Q!
    </h1>
  </div>
</DocumentFragment>
`;
